pkgs/dart_mcp/test/resources_support_test.dart

```diff
@@ -136,6 +136,81 @@ void main() {
     await environment.shutdown();
   });
 
+  test('resource change notifications are throttled', () async {
+    final environment = TestEnvironment(
+      TestMCPClient(),
+      (c) => TestMCPServerWithResources(channel: c),
+    );
+    await environment.initializeServer();
+
+    final serverConnection = environment.serverConnection;
+    final server = environment.server;
+
+    final resourceListChangedQueue = StreamQueue(
+      serverConnection.resourceListChanged,
+    );
+
+    final resources = [
+      for (var i = 0; i < 5; i++) Resource(name: '$i', uri: 'foo://$i'),
+    ];
+    for (var resource in resources) {
+      server.addResource(
+        resource,
+        (_) => ReadResourceResult(
+          contents: [
+            TextResourceContents(uri: resource.uri, text: resource.name),
+          ],
+        ),
+      );
+    }
+
+    // Should get exactly two notifications even though we have more resources,
+    // one initial notification and one after the throttle delay.
+    await resourceListChangedQueue.take(2);
+    expect(
+      resourceListChangedQueue.hasNext.timeout(
+        const Duration(milliseconds: 10),
+        onTimeout: () => false,
+      ),
+      completion(false),
+    );
```

Comment:
I don't think it adds much safety to check for no-events within a bound window of time. We're limited by overall test runtime, but we can make a stronger expectation that no further events are emitted here. How about

```suggestion
    expect(
      resourceListChangedQueue.hasNext,
      doesNotComplete,
    );
```

Reply:
Updated this to `completion(false)` with a zero delay for the throttling and `pumpEventQueue` to ensure delivery of events.